WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift

```diff
@@ -22,10 +22,10 @@ public final class SingleProductScreen: ScreenObject {
         return try ProductsScreen()
     }
 
-    public func verifyProductOnSingleProductScreen(name: String, price: String, status: String) throws -> Self {
-        XCTAssertTrue(try app.getTextVisibilityCount(text: status) == 1, "Stock status does not exist!")
-        XCTAssertTrue(try app.getTextVisibilityCount(text: price) == 1, "Price does not exist!")
-        XCTAssertTrue(app.textViews[name].isFullyVisibleOnScreen(), "Product name does not exist!")
+    public func verifyProductOnSingleProductScreen(products: [ProductData]) throws -> Self {
+        XCTAssertTrue(try app.getTextVisibilityCount(text: products[0].stock_status) == 1, "Stock status does not exist!")
+        XCTAssertTrue(try app.getTextVisibilityCount(text: products[0].regular_price) == 1, "Price does not exist!")
+        XCTAssertTrue(app.textViews[products[0].name].isFullyVisibleOnScreen(), "Product name does not exist!")
```

Comment:
Since this method tests only a single product page, I'd expect it to accept a single product as a parameter (the rest of the array is not used anyway):

```suggestion
    public func verifyProductOnSingleProductScreen(product: ProductData) throws -> Self {
        XCTAssertTrue(try app.getTextVisibilityCount(text: product.stock_status) == 1, "Stock status does not exist!")
        XCTAssertTrue(try app.getTextVisibilityCount(text: product.regular_price) == 1, "Price does not exist!")
        XCTAssertTrue(app.textViews[product.name].isFullyVisibleOnScreen(), "Product name does not exist!")
```

Also, I find it very helpful (especially when there's need to debug things on CI), to provide the expected (and actual, if applicable) values in asserting messages.

Reply:
Makes sense, I'll remove the array and will use the first item on it, will update the message as well.